src/celestia/celestiacore.cpp

```diff
@@ -4745,22 +4745,25 @@ View* CelestiaCore::getViewByObserver(const Observer *obs) const
     return nullptr;
 }
 
-Image CelestiaCore::captureImage() const
+void CelestiaCore::getCaptureInfo(std::array<int, 4>& viewport, celestia::PixelFormat& format) const
 {
-    // Get the dimensions of the current viewport
-    array<int, 4> viewport;
-    getRenderer()->getViewport(viewport);
-
-    PixelFormat format = renderer->getPreferredCaptureFormat();
+    renderer->getViewport(viewport);
+    format = renderer->getPreferredCaptureFormat();
+}
 
-    Image image(format, viewport[2], viewport[3]);
-    if (!renderer->captureFrame(viewport[0], viewport[1],
-                                viewport[2], viewport[3],
-                                format, image.getPixels()))
+bool CelestiaCore::captureImage(std::uint8_t* buffer,
```

Comment:
std::byte?

Reply:
Using `std::byte` would introduce a bit of friction with both Celestia's `Image` class and Qt's `QImage` (which are the relevant destinations here), which both use `std::uint8_t`/`unsigned char` to store their data. It might be something to consider doing as a larger refactoring of the `Image` class though (although interacting with `QImage` would still end up needing `reinterpret_cast`).

Reply:
okay, let's refactor here &  Image later